### DISCUSSION TITLE
Cloud_Firestore-Error (Flutter-Web) => [cloud_firestore/unavailable] Failed to get document because the client is offline.

### DISCUSSION THREAD

**Discussion by IB-Scherr** [firebase/flutterfire#12846] [Q&A]
Hi Guys,

the following snippet =>

    _final docRef = FirebaseFirestore.instance.collection("User").doc("JohnDoe");
    docRef.get().then(
            (DocumentSnapshot doc) {
          print("DATA");
        },
        onError: (e) => print("Error getting document: $e"),);_

causes this error after a while =>

_Error getting document: [cloud_firestore/unavailable] Failed to get document because the client is offline._

And nothing happens. Additionally I couldn't see any request via developer tools.

I followed the integration manual from different sources via the Firebase CLI but without any success. The firebase_options-file was generated with the corresponding parameters. And I could see in my firebase project backend that the connection happens successfully. 

The requested collection and document was created =>

{
  "documents": [
    {
      "name": "projects/.../databases/db-v1/documents/User/JohnDoe",
      "fields": {
        "Name": {
          "stringValue": "John"
        },
        "FamilyName": {
          "stringValue": "Doe"
        }
      },
      "createTime": "2024-05-29T15:31:52.495514Z",
      "updateTime": "2024-05-29T15:31:52.495514Z"
    }
  ]
}

It was possible for me to use the authentification module without any issues within the app.

The main entry point was initialized like proposed =>

_Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp(
    options: DefaultFirebaseOptions.web,
  );_

Flutter + all packages were updated.

After 5min debugging I'm ending up here. 

Does anybody know what is missing ?

++ Yesterday I tested it locally with the emulator. There it works great. 
++ And there aren't any rules set to prevent any access on Firebase.
=> The problem is still unsolved.

**Comment by magrippa27** [comment]
Same issue here

**Comment by Chris-onoh** [comment]
type shii, i think its a web only problem, imma check it on mobile, it annoying AF

**Comment by IB-Scherr** [comment]
I stopped using the web deploy for development purposes because I was also struggling with some other plugins. On the mobile it's finally working. I will ignore this and close the ticket.
